src/commands/reword.rs

```diff
@@ -0,0 +1,314 @@
+
+    Ok((repo, dag))
+}
+
+/// Turn a list of ref-ish strings into a list of Commits.
+fn resolve_commits_from_hashes<'repo>(
+    repo: &'repo Repo,
+    dag: &mut Dag,
+    effects: &Effects,
+    hashes: Vec<String>,
+) -> eyre::Result<Option<Vec<Commit<'repo>>>> {
+    let hashes = if hashes.is_empty() {
+        vec!["HEAD".to_string()]
+    } else {
+        hashes
+    };
+
+    let commits = resolve_commits(effects, repo, dag, hashes)?;
+    let commits = match commits {
+        ResolveCommitsResult::Ok { commits } => commits,
+        ResolveCommitsResult::CommitNotFound { commit: hash } => {
+            writeln!(effects.get_output_stream(), "Commit not found: {}", hash)?;
+            return Ok(None);
+        }
+    };
+    Ok(Some(commits))
+}
+
+/// The result of building the reword message.
+#[must_use]
+pub enum BuildRewordMessageResult {
+    /// The reworded message was built successfully.
+    Succeeded {
+        /// The reworded messages for each commit.
+        messages: HashMap<NonZeroOid, String>,
+    },
+
+    /// The reworded message matches the original message.
+    IdenticalMessage,
+
+    /// The reworded message was empty.
+    EmptyMessage,
+
+    /// Misc failure.
+    Failed {
+        /// The failue message, if any.
+        message: Option<String>,
+    },
+}
+
+/// Builds the message(s) that will be used for rewording. These are mapped from each commit's
+/// NonZeroOid to the relevant message.
+fn build_messages(
+    messages: &[String],
+    commits: &[Commit],
+) -> eyre::Result<BuildRewordMessageResult> {
+    let message = messages.join("\n\n").trim().to_string();
+
+    // TODO(maybe?) stdin?
+
+    let (message, load_editor) = if message.is_empty() {
+        let message_for_editor = match commits.len() {
+            1 => {
+                // FIXME lots of error checking to do
+                let commit = commits.first().unwrap();
+                let msg = commit.get_message_raw()?;
+                let msg = msg.into_string();
+                match msg {
+                    Ok(msg) => msg,
+                    _ => {
+                        return Ok(BuildRewordMessageResult::Failed {
+                            message: Some(String::from(
+                                "Reword: Error decoding original commit message!",
+                            )),
+                        })
+                    }
+                }
+            }
+            len => {
+                // TODO build a bulk edit message for multiple commits
+                format!("Enter a commit message to apply to {} commits", len)
+            }
+        };
+        (message_for_editor, true)
+    } else {
+        (message, false)
+    };
+
+    let message = if load_editor {
+        // Editor::edit will only return None if saving is required; b/c we've disabled the save
+        // requirement, we don't need to worry about it.
+        let edited_message = Editor::new()
+            .require_save(false)
```

Comment:
It's just a configurable behavior of the `dialoguer::Editor`. By default, if you close the editor w/o saving, this would return `None`. I don't think it's relevant to care about the file being saved, only if the resulting output message is the same as the input message, so I disabled this requirement. I'll update the comment to make that more clear.